src/github.com/couchbase/sync_gateway/base/rlimit.go

```diff
@@ -4,25 +4,97 @@ package base
 
 import "syscall"
 
-func SetMaxFileDescriptors(maxFDs uint64) (uint64, error) {
+// Set Max File Descriptor limits
+//
+// Background information:
+//
+// - SG docs
+//   http://developer.couchbase.com/documentation/mobile/1.1.0/develop/guides/sync-gateway/os-level-tuning/max-file-descriptors/index.html
+// - Related SG issues
+//   https://github.com/couchbase/sync_gateway/issues/1083
+// - Hard limit vs Soft limit
+//   http://unix.stackexchange.com/questions/29577/ulimit-difference-between-hard-and-soft-limits
+func SetMaxFileDescriptors(requestedSoftFDLimit uint64) (uint64, error) {
+
+	requiresUpdate, recommendedSoftFDLimit, err := getSoftFDLimit(requestedSoftFDLimit)
+	if err != nil {
+		return 0, err
+	}
+
+	// No call to Setrlimit required, because the requested soft limit is lower than current soft limit
+	if !requiresUpdate {
+		return 0, nil
+	}
+
+	// Get existing limits
 	var limits syscall.Rlimit
 	if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &limits); err != nil {
-		return maxFDs, err
+		return 0, err
```

Comment:
Is this call to Getrlimit still needed, since we just made the same call inside getSoftFDLimit?